flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -178,7 +150,14 @@ private void updateForReconciliationError(FlinkDeployment flinkApp, String err)
     @Override
     public Optional<FlinkDeployment> updateErrorStatus(
             FlinkDeployment flinkApp, RetryInfo retryInfo, RuntimeException e) {
-        LOG.warn("TODO: handle error status");
-        return Optional.empty();
+        LOG.warn(
```

Comment:
The log messages are as expected, but maybe we should also capture the retry info in the status? And add an event when retries are exhausted? Otherwise, that information is only transient in the operator SDK?